File: apps/examples/tenant-xpto/food-market-01-service/src/main/java/com/darueira/foodmarket/service01/domain/port/in/FoodTradingUseCase.java
package com.darueira.foodmarket.service01.domain.port.in;

import com.darueira.foodmarket.service01.domain.model.CreateFoodTradingCommand;
import com.darueira.foodmarket.service01.domain.model.FoodTrading;
import com.darueira.foodmarket.service01.domain.model.FoodTradingEvent;
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;

import java.util.List;

public interface FoodTradingUseCase {
    FoodTrading createTrading(CreateFoodTradingCommand command);
    List<FoodTrading> listTradings();
    SseEmitter subscribeStream();
    void processIncomingEvent(FoodTradingEvent event);
}

File: apps/examples/tenant-xpto/food-market-02-service/src/main/kotlin/com/darueira/foodmarket/service02/domain/port/in/FoodTradingUseCase.kt
package com.darueira.foodmarket.service02.domain.port.`in`

import com.darueira.foodmarket.service02.domain.model.CreateFoodTradingCommand
import com.darueira.foodmarket.service02.domain.model.FoodTrading
import com.darueira.foodmarket.service02.domain.model.FoodTradingEvent
import io.smallrye.mutiny.Multi

interface FoodTradingUseCase {
    fun createTrading(command: CreateFoodTradingCommand): FoodTrading
    fun listTradings(): List<FoodTrading>
    fun subscribeStream(): Multi<FoodTrading>
    fun processIncomingEvent(event: FoodTradingEvent)
}
